src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -510,7 +510,10 @@ def _add_role_assignment(cli_ctx, role, service_principal, delay=2):
         hook.add(message='Waiting for AAD role to propagate', value=0.1 * x, total_val=1.0)
         try:
             # TODO: break this out into a shared utility library
-            create_role_assignment(cli_ctx, role, service_principal)
+            if not scope:
+                create_role_assignment(cli_ctx, role, service_principal)
+            else:
+                create_role_assignment(cli_ctx, role, service_principal, scope=scope)
```

Comment:
Combine these into `create_role_assignment(cli_ctx, role, service_principal, scope=scope)`.

Reply:
It doesn't look like the `if not scope:` test is necessary, I agree. Is the default value for `scope` not `None` in that function?

Reply:
That should work; [create_role_assignment](https://github.com/Azure/azure-cli/blob/dev/src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py#L1206) has the following signature:
`def create_role_assignment(cli_ctx, role, assignee, resource_group_name=None, scope=None):`

If scope is None, it will just set that.